### PR TITLE
waypoints 0.05: implement goto functionality

### PR DESCRIPTION
**Pull request by pavelmachek**
Update libraries, refactor and implement goto functionality

### CODE REVIEW

File: apps/waypoints/waypoints.app.js [hunk 1]
```diff
@@ -219,10 +490,85 @@ function showNumpad(text, key_, callback) {
...
+  g.clear();
+  la.render();
+}
+
+function showCard() {
+  var menu = {
+    "" : {title : "Select WP"},
+    "< Back" : mainMenu
+  };
+  if (Object.keys(wp).length==0) Object.assign(menu, {"No WPs":""});
```
Comment (bobrippling) [inline apps/waypoints/waypoints.app.js:543]: Is `wp` always an array? If so we could:
```suggestion
  if (wp.length==0) Object.assign(menu, {"No WPs":""});
```

File: apps/waypoints/waypoints.app.js [hunk 2]
```diff
@@ -219,10 +490,85 @@ function showNumpad(text, key_, callback) {
...
+  else {
+    wp.forEach((val, card) => {
+      const name = wp[card].name;
```
Comment (bobrippling) [inline apps/waypoints/waypoints.app.js:546]: Here we could save a bit of code:
```suggestion
      const name = val.name;
```

**Review by bobrippling — COMMENTED** [review]
Looks good, two minor suggestions

**Comment by pavelmachek** [comment]
Yeah, there's ton more optimalizations/cleanups that could be done. I'd prefer this to be applied, then I can take your patches as a separate commit, do more work on top of that, and submit it as new version after some testing?

**Comment by pavelmachek** [comment]
Thank you, it makes my patch flow easier.

**Pull request merged**